src/NewTools-Debugger/Context.extension.st

```diff
@@ -0,0 +1,20 @@
+Extension { #name : #Context }
+
+{ #category : #'*NewTools-Debugger' }
+Context >> findFailedAssertionContextInSenderChain [
+
+	^ (self findContextSuchThat: [ :ctx | 
+		   (ctx hasTemporaryVariableNamed: #actual) and: [ 
+			   ctx hasTemporaryVariableNamed: #expected ] ]) ifNil: [ 
```

Comment:
It is very fragile logic. The names #actual and #expected is kind of SUnit pattern to name variables in the tests. 
For example your logic will not fail for test like:
```Smalltalk
actual := 10 + 20.
expected := 2.
self assert: actual / 0 equals: expected
```
I did not check what debugger will show in that case. But the logic looks wrong.

Reply:
Another example of why relying on var names is a bad idea:
```Smalltalk
assert: actualNumber closeTo: expectedNumber
```
There will be no assertion tab in the debugger for such assertion failure

Reply:
Two things:
- the debugger only look for those when it has insurance it is a test failure, so somewhere down the stack there is a test failure (just for information)
- it is better than before (even in GT) where those values were fetched from magic indexes in the test failure context
I am open to suggestions about how to do better though :)

Reply:
> It is very fragile logic. The names #actual and #expected is kind of SUnit pattern to name variables in the tests.
> For example your logic will not fail for test like:
> 
> ```smalltalk
> actual := 10 + 20.
> expected := 2.
> self assert: actual / 0 equals: expected
> ```
> 
> I did not check what debugger will show in that case. But the logic looks wrong.

Actually this produces a divide by zero exception, so as the debugger is not in a test failure context it is not going to try displaying the diff view.